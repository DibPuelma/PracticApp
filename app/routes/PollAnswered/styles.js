import {
  StyleSheet,
  Dimensions
} from 'react-native'
module.exports = StyleSheet.create({
  container: {
    backgroundColor: '#F6F6F6',
    width: Dimensions.get('window').width,
    height: Dimensions.get('window').height,
    justifyContent: 'center',
    alignItems: 'center'
  },
  card:Â {
    margin: 20,
    borderWidth: 0.5,
    borderRadius: 5,
    borderColor: '#E6E6E6',
    backgroundColor: '#E6E6E6',
    elevation: 3
  },
  title: {
    marginTop: 20,
    marginBottom: 20,
    textAlign: 'center',
    fontSize: 30
  },
  normalText: {
    textAlign: 'center',
    fontSize: 20,
    marginBottom: 30
  }
});
