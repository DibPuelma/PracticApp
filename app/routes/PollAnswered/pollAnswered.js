import React, { Component } from 'react';
import {
  View,
  Button,
  Text,
  BackAndroid
} from 'react-native';
import styles from './styles';

import backButtonHandler from '../../lib/backButtonHandler';

export default class PollAnswered extends Component {
  constructor(props){
    super(props);
    this.singletonBackButtonHandler = backButtonHandler.getInstance();
    this._accept = this._accept.bind(this);
  }

  componentWillMount(){
    this._addBackEvent();
  }

  componentWillUnmount(){
    this._removeBackEvent();
  }

  _addBackEvent() {
    BackAndroid.addEventListener('hardwareBackPress', this._accept);
    this.singletonBackButtonHandler.addFunction(this._accept);
  }

  _removeBackEvent() {
    BackAndroid.removeEventListener('hardwareBackPress', this._accept);
    this.singletonBackButtonHandler.removeFunction(this._accept);
  }

  render(){
    return(
      <View style={styles.container}>
      <View style={styles.card}>
        <Text style={styles.title}> ¡Gracias por contestar la encuesta! </Text>
        <Text style={styles.normalText}> Tu opinión servirá para que cada vez te atiendan mejor </Text>
        <Button onPress={() => this._accept()} title="Aceptar" color="#841584"
        accessibilityLabel="Acepta"
        />
      </View>
      </View>

      // <View>
      // <Text> Gracias por contestar la encuesta </Text>
      // <Text> Tienda: {this.props.pollData.store} </Text>
      // <Text> Vendedor: {this.props.pollData.employeeId} </Text>
      // {this.props.pollAnswers.stars.map((answer, i) => (<Text key={i}> Pregunta {i+1}: {this.props.pollAnswers.stars[i]} estrellas </Text>))}
      // <Text> Comentario: {this.props.pollAnswers.comment} </Text>
      // <Button onPress={() => this._accept()} title="Aceptar" color="#841584"
      // accessibilityLabel="Acepta"
      // />
      // </View>
    );
  }
  _accept(){
    this.props.navigator.replace({id:'scanner'});
    return true;
  }
}
