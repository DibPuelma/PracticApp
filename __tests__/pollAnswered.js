import 'react-native';
import React from 'react';
import PollAnswered from '../app/routes/PollAnswered/pollAnswered'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <PollAnswered />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
