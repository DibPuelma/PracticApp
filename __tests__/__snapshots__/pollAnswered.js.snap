exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F6F6F6",
      "height": 1334,
      "justifyContent": "center",
      "width": 750,
    }
  }>
  <View
    style={
      Object {
        "backgroundColor": "#E6E6E6",
        "borderColor": "#E6E6E6",
        "borderRadius": 5,
        "borderWidth": 0.5,
        "elevation": 3,
        "margin": 20,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 30,
          "marginBottom": 20,
          "marginTop": 20,
          "textAlign": "center",
        }
      }>
       ¡Gracias por contestar la encuesta! 
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 20,
          "marginBottom": 30,
          "textAlign": "center",
        }
      }>
       Tu opinión servirá para que cada vez te atiendan mejor 
    </Text>
    <View
      accessibilityComponentType="button"
      accessibilityLabel="Acepta"
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}>
      <View
        style={
          Array [
            Object {},
          ]
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#0C42FD",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#841584",
              },
            ]
          }>
          Aceptar
        </Text>
      </View>
    </View>
  </View>
</View>
`;
